envoy/config/filter/http/jwt_authn/v2/config.proto

```diff
@@ -0,0 +1,197 @@
+
+syntax = "proto3";
+
+package envoy.config.filter.http.jwt_authn.v2;
+
+import "envoy/api/v2/route/route.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/wrappers.proto";
+import "validate/validate.proto";
+
+// JSON Web Token (JWT) token format for authentication is defined
+// `here <https://tools.ietf.org/html/rfc7519>`_.
+// See `OAuth2.0 <https://tools.ietf.org/html/rfc6749>`_ and
+// `OIDC1.0 <http://openid.net/connect>`_ for the authentication flow.
+//
+// Example,::
+//
+//     issuer: https://example.com
+//     audiences:
+//     - bookstore_android.apps.googleusercontent.com
+//       bookstore_web.apps.googleusercontent.com
+//     jwks_uri: https://example.com/.well-known/jwks.json
+//
+// [#not-implemented-hide:]
+message JSONWebToken {
+  // Identifies the principal that issued the JWT. See
+  // `here <https://tools.ietf.org/html/rfc7519#section-4.1.1>`_.
+  // Usually a URL or an email address.
+  //
+  // Example: https://securetoken.google.com
+  // Example: 1234567-compute@developer.gserviceaccount.com
+  //
+  string issuer = 1;
+
+  // The list of JWT `audiences
+  // <https://tools.ietf.org/html/rfc7519#section-4.1.3>`_. that are allowed to
+  // access. A JWT containing any of these audiences will be accepted.
+  //
+  // Example::
+  //
+  //     audiences:
+  //     - bookstore_android.apps.googleusercontent.com
+  //       bookstore_web.apps.googleusercontent.com
+  //
+  repeated string audiences = 2;
+
+  // URL of the provider's public key set to validate signature of the
+  // JWT. See `OpenID
+  // Discovery
+  // <https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata>`_
+  //
+  // Example: https://www.googleapis.com/oauth2/v1/certs
+  string jwks_uri = 3;
+
+  // This field is specific for Envoy proxy implementation.
+  // It is the cluster name in the Envoy "cluster_manager" config section.
+  // In order for Envoy to call "jwks_uri", its host has to be specified
+  // as a "cluster" in the config for each jwks_uri.
+  string jwks_uri_cluster = 4;
+
+  // Duration after which the cached public key should be expired. The
+  // system wide default is applied if no duration is explicitly
```

Comment:
Where does the system wide default come from?

Reply:
will update to
 hard-coded default of 5 minutes